src/TodoFilter.js

```diff
@@ -0,0 +1,39 @@
+import {FilterDetails} from './constants.js';
+
+export default function TodoFilter($todoFilter, data, selected, filterItems) {
+  this.$todoFilter = $todoFilter;
+  this.data = data;
+  this.selected = selected;
+
+  this.init = () => {
+    this.render();
+    this.bind();
+  };
+
+  this.bind = () => {
+    this.$todoFilter.addEventListener('click', ({target}) => {
+      FilterDetails.filter(({type}) => {
+        if (target.classList.contains(type)) {
+          this.selected = type;
+          filterItems(this.selected);
+        }
+      });
+      this.render();
+    });
+  };
+
+  this.render = () => {
+    let result = '';
+    FilterDetails.map(({type, text}) => {
+      result += `
+        <li>
+          <a class="${type} ${this.selected == type? 'selected': ''}" href="#${type}">${text}</a>
+        </li>
+        `;
+    });
+
+    this.$todoFilter.innerHTML = result;
```

Comment:
map을 forEach 처럼 사용하고 있네요! map은 새로운 값을 반환해줘야합니다.
```suggestion
    this.$todoFilter.innerHTML = FilterDetails.map(({type, text}) => `
        <li>
          <a class="${type} ${this.selected == type? 'selected': ''}" href="#${type}">${text}</a>
        </li>
    `).join('');
```
즉, 배열의 요소({ type, text })를 string으로 변환하여 새로운 배열을 만든 후에 이어주는거죠!

예를 들자면 다음과 같습니다.
```js
const originArr = [1, 2, 3, 4, 5];
const arr1 = originArr.map(v => v * 10);
const arr2 = originArr.map(v => `${v}${v}`).join('/');
console.log(arr1); // [10, 20, 30, 40, 50]
console.log(arr2); // "11/22/33/44/55"
```

Reply:
제가 맵을 이상하게 사용하고 있었네요.. 핃백감사합니다!!!